src/components/Searchbar/SearchBox/SearchBox.tsx

```diff
@@ -31,185 +31,189 @@ type SearchBoxProps = {
+    onMouseMove,
+    hasFocus,
+  }) => {
+    const { channels, videos, loading } = useSearchResults({ searchQuery })
+    const { recentSearches, deleteRecentSearch } = usePersonalDataStore((state) => ({
+      recentSearches: state.recentSearches,
+      deleteRecentSearch: state.actions.deleteRecentSearch,
+    }))
+    const containerRef = useRef<HTMLDivElement>(null)
+    const topRef = useRef(0)
+    const [visualViewportHeight, setVisualViewportHeight] = useState(window.visualViewport.height)
 
-  // Calculate searchbox height whether keyboard is open or not
-  useEffect(() => {
-    const debouncedVisualViewportChange = debounce(() => {
-      setVisualViewportHeight(window.visualViewport.height)
-    }, 100)
-    window.visualViewport.addEventListener('resize', debouncedVisualViewportChange)
+    // Calculate searchbox height whether keyboard is open or not
+    useEffect(() => {
+      const debouncedVisualViewportChange = debounce(() => {
+        setVisualViewportHeight(window.visualViewport.height)
+      }, 100)
+      window.visualViewport.addEventListener('resize', debouncedVisualViewportChange)
 
-    return () => {
-      window.visualViewport.removeEventListener('resize', debouncedVisualViewportChange)
-    }
-  }, [])
-
-  const scrollToSelectedItem = useCallback(
-    (top: number, title?: string | null) => {
-      onSelectItem(title)
-      if (!containerRef.current) {
-        return
-      }
-      const { offsetHeight } = containerRef.current
-      if (selectedItem === 0 || top < offsetHeight) {
-        containerRef?.current?.scrollTo(0, 0)
+      return () => {
+        window.visualViewport.removeEventListener('resize', debouncedVisualViewportChange)
       }
-      if (top >= offsetHeight + (top < topRef.current ? -250 : -50)) {
-        containerRef?.current?.scrollTo(0, top + (top < topRef.current ? -50 : -250))
-      }
-      topRef.current = top
-    },
-    [onSelectItem, selectedItem]
-  )
+    }, [])
 
-  const generatePlaceholders = useMemo(() => {
-    const min = 20
-    const max = 80
-    const placeholderItems = Array.from({ length: 6 }, () => ({ id: undefined }))
-    return placeholderItems.map((_, idx) => {
-      const generatedWidth = Math.floor(Math.random() * (max - min)) + min
-      return (
-        <PlaceholderWrapper key={`placeholder-${idx}`}>
-          <SkeletonAvatar width="32px" height="32px" rounded />
-          <SkeletonLoader width={`${generatedWidth}%`} height="16px" />
-        </PlaceholderWrapper>
-      )
-    })
-  }, [])
+    const scrollToSelectedItem = useCallback(
+      (top: number, title?: string | null) => {
+        onSelectItem(title)
+        if (!containerRef.current) {
+          return
+        }
+        const { offsetHeight } = containerRef.current
+        if (selectedItem === 0 || top < offsetHeight) {
+          containerRef?.current?.scrollTo(0, 0)
+        }
+        if (top >= offsetHeight + (top < topRef.current ? -250 : -50)) {
```

Comment:
Lets avoid magic values here like the -250 and -50 if they mean something lets have them as a variable like TOP_BAR_HEIGHT or so